backend/matching-service/app/src/matching/matching.interfaces.ts

```diff
@@ -0,0 +1,17 @@
+import { Socket } from "socket.io";
+
+export interface Preferences {
+  difficulty: string[];
+  category: string[];
+}
+
+export interface UserSocket {
+  user: number;
```

Comment:
what do u think about naming this as `userId` for clarity?

Reply:
initially i was gonna pass the whole user object in, so we don't have to query for it again, but the object is not available yet so im using a number first

Reply:
ahh okay sounds good